src/main/java/fr/sncf/osrd/train/Train.java

```diff
@@ -53,10 +53,10 @@ public static Train create(
                 0
         );
 
-        ActivateRoute.trainCreation(sim, initialState);
 
         var trainCreatedChange = new TrainCreatedChange(sim, schedule, initialState);
         var train = trainCreatedChange.apply(sim);
+        ActivateRoute.trainCreation(sim, initialState);
```

Comment:
This seems weird. Why the route should be activated after creating the train?

Reply:
We need the train to be created to request a reservation here: https://github.com/DGEXSolutions/osrd-core/pull/130/files#diff-8592b45448bfbb047d9f36930ddfc0298530bc3489755028ea72a3b71af1fb48R68